src/common/util/include/openvino/util/env_util.hpp

```diff
@@ -31,5 +31,12 @@ int32_t getenv_int(const char* env_var, int32_t default_value = -1);
 /// \param default_value The value to return if the environment variable is not set.
 /// \return Returns the boolean value of the environment variable.
 bool getenv_bool(const char* env_var, bool default_value = false);
+
+/// \brief Splits a string using by a single character delimiter
+///        without modifying the original string
+/// \param str String to split
+/// \param delimiter Delimiter to use for splitting
+/// \return Returns an unordered set of split strings
+std::unordered_set<std::string> split_by_delimiter(const std::string& str, char delimiter);
```

Comment:
Move it to "common_util.hpp"
```suggestion
std::vector<std::string> split(const std::string& str, const std::string& delimiter = ",");
```

Return value can remove some split strings, it can be always created from vector.
And use string delimiter as it can provide more complex option for split the text